### PR TITLE
Match Genius paths

### PR DESCRIPTION
**Pull request by Insprill**
Currently, we have our own schema for paths, with Genius paths being passed in query parameters. This makes it complicated to set automatic redirects from Genius to Intellectual and adds unnecessary complexity to the code.

This will need some testing to ensure that there are no edge cases that break the links and that no links were missed.
~~We also need a way to return a proper 404 page, since the lyric pattern now matches all paths. We might be able to only match paths that end in `-lyrics`, or we'll have to bail and return our own 404 if genius returns a 404.~~

We should also try to coordinate the release of this with LibRedirect to minimize the damage from this.

### CODE REVIEW

File: src/artist.rs [hunk 1]
```diff
@@ -13,7 +12,7 @@ use crate::templates::template;
 static GENIUS_IMAGE_URL: &str = "https://images.genius.com/";
 static GENIUS_BASE_PATTERN: Lazy<Regex> = lazy_regex!(r#"https?://\w*.?genius\.com/"#);
 static GENIUS_ALBUMS_PATTERN: Lazy<Regex> = lazy_regex!(r#"https?://\w*.?genius\.com/albums/"#);
-static GENIUS_ARTIST_PATTERN: Lazy<Regex> = lazy_regex!(r#"https?://\w*.?genius\.com/artists/"#);
+static GENIUS_ARTIST_PATTERN: Lazy<Regex> = lazy_regex!(r#"https?://\w*.?genius\.com/"#);
```
Comment (Insprill) [inline src/artist.rs]: test
